### PR TITLE
[Feature]: Account for existing rules

### PR DESCRIPTION
**Pull request by r-scheele**
Updated the write_rego_file function to append a rule if the auth.rego file isn't empty.

### CODE REVIEW

**Review by Youngestdev — APPROVED** [review]

File: app/utils/write_rego.py [hunk 1]
```diff
@@ -23,14 +22,27 @@ def write_to_file(rule: RequestObject) -> dict:
...
+
+    # read the file
+    with open(file_path, "r") as file:
+        data = file.read() if operation == "write" else rule["old_state"]
+        result = (
+            data + build_rego(rule["rules"])
+            if data
+            else initiate_rule + build_rego(rule["rules"])
+        )
+
```
Comment (Youngestdev) [inline app/utils/write_rego.py:35]: We should be checking if the file exists first before attempting to open it. This breaks the POST route if the repo is empty.

**Review by Youngestdev — CHANGES_REQUESTED** [review]

File: app/utils/write_rego.py [hunk 2]
```diff
@@ -26,28 +26,24 @@ def write_to_file(rule, operation: str = "write") -> dict:
...
 
-    return {"status": "error", "message": "Policy is invalid"}
+    with open(file_path, "w") as file:
+        file.write(result)
+    # Update GitHub
+    github.push()
+    return {
+        "status": "success",
+        "message": "Policy successfully written to file",
+        "old_state": old_rule,
```
[the remark this answers is not in the document]
Comment (r-scheele) [reply app/utils/write_rego.py:45]: Alright. That should be all right?

File: app/server/api.py [hunk 3]
```diff
@@ -18,18 +18,13 @@ async def write_policy(rego_rule: RequestObject, database=Depends(get_db)) -> di
...
-            "state": rego_rule,
-        }
-    else:
-        raise HTTPException(status_code=400, detail=response["message"])
+    write_to_file(rego_rule, operation="write")
+
+    return {
+        "status": 200,
+        "message": "Policy created successfully",
+        "state": rego_rule,
```
Comment (Youngestdev) [inline app/server/api.py:26]: Remove this.

File: app/server/github.py [hunk 4]
```diff
@@ -50,7 +50,7 @@ def push(self):
             if remotes[0].name != "origin":
                 repo.create_remote("origin", target_url)
             origin = repo.remote(name="origin")
-            origin.fetch()
+            origin.pull()
```
Comment (Youngestdev) [inline app/server/github.py:53]:
```suggestion
            origin.fetch()
```

`.fetch()` now works.

**Review by Youngestdev — CHANGES_REQUESTED** [review]

File: app/server/api.py (hunk 3 above)
Comment (Youngestdev) [inline app/server/api.py:26]:
```suggestion
```

Comment (r-scheele) [reply app/server/api.py:26]: Shouldn't return a response?

Comment (Youngestdev) [reply app/server/api.py:26]: See the suggested change. 

Also replace .pull() with .fetch.

**Pull request merged**